src/components/vocabulary/AddVocabularyToProject.tsx

```diff
@@ -54,34 +58,36 @@ const AddVocabularyToProject: React.FC<AddVocabularyToProjectProps> = ({
   }, [data, project.vocabularyContexts]);
 
   const addVocabularyToProject = async (vocabulary: BaseVocabularyData) => {
-    setIsWaiting(true);
+    setBusy(true);
     notifyPromise(
       addVocabularyToExistingProject(vocabulary, project),
       formatProjectCreationMessage()
     )
       .then((instanceID) => {
-        setIsWaiting(false);
+        setBusy(false);
         queryClient.invalidateQueries(["projectsID", instanceID]);
       })
-      .catch(() => setIsWaiting(false));
+      .catch(() => {
+        setBusy(false);
+      });
   };
 
   const createVocabularyToProject = async (
     vocabulary: AddVocabularyPayload
   ) => {
-    setIsWaiting(true);
+    setBusy(true);
     notifyPromise(
       createVocabularyToExistingProject(vocabulary, project),
       formatProjectCreationMessage()
     )
       .then((instanceID) => {
-        setIsWaiting(false);
+        setBusy(false);
         queryClient.invalidateQueries(["projectsID", instanceID]);
       })
-      .catch(() => setIsWaiting(false));
+      .catch(() => setBusy(false));
```

Comment:
I personally like to catch with the error message - at least a `.catch((e) => {console.error(e); ...` - this helps me figure out if and specifically where something went wrong.

Reply:
Of course that is the better way. However the logging is unfortunately not present anywhere in the app, so I left it blank for now.